### PR TITLE
Integrating with Amazon RDS database

### PR DESCRIPTION
**Pull request by MChartier**
Adding database.go file defining a function that establishes a connection with a remote database
given an endpoint and password defined through environment variables.

Updating application.go to include a handler on the "/counter" route that serves up the latest values
of an integer counter and then increments the value in the database.

We are using password authentication to connect to the database instance because IAM authentication
is not available for the small instance size that we have provisioned in AWS. IAM authentication will also
not be supported if and when we switch to using a serverless deployment of Aurora, so we may need to continue
connecting in this way. Environment variables are used to store the endpoint and password so that we can configure
different environments to connect to different instances of the database and to ensure that we do not check in
database credentials to source control.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,11 +3,16 @@ Web API service for Matterhorn implemented with Go.
 
 ## Setup
 
+### Database
+MatterhornApiService expects to connect to a MySQL-compatible database instance. You will need to set some environment variables to define how to connect.
+`MATTERHORN_DB_ENDPOINT` - Specifies the host name of the database instance, including the port. E.g., http://mydatabase.com:3306.
+`MATTERHORN_DB_PASSWORD` - Specifies the password used to connect to the database instance. User name is expected to be "admin" for now.
```
Comment (chriswr95) [inline README.md:9]: Should we provide the location for the password, ie check slack or ask Matt

File: application.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		panic("Failed to connect to DB! Error: " + err.Error())
+	}
+
+	if db == nil {
+		panic("DB is nil immediately after assignment!")
+	}
+
+	defer func() {
+		log.Println("Closing db...")
+		// db.Close()
```
Comment (chriswr95) [inline application.go]: Why is db.Close() commented out?

**Review by chriswr95 — APPROVED** [review]

File: application.go (hunk 2 above)
Comment (MChartier) [reply application.go]: No good reason, good catch.

File: README.md (hunk 1 above)
Comment (MChartier) [reply README.md:9]: Nah, let's keep track of those sorts of operational logistics independently of the code.

In practice going forward you (or anybody who clones this repo) will be able to create a compatible database instance. The fact that we currently intend to use this with a particular Amazon RDS database instance is incidental to the implementation.

**Pull request merged**